src/services/rfqm_service.ts

```diff
@@ -804,68 +804,80 @@ export class RfqmService {
-                                decodedFillLog: JSON.stringify(decodedLog),
-                            };
-                        } else {
-                            jobStatus = isTxConfirmed
-                                ? RfqmJobStatus.FailedRevertedConfirmed
-                                : RfqmJobStatus.FailedRevertedUnconfirmed;
-                            submissionsMap[r.transactionHash].status = isTxConfirmed
-                                ? RfqmTransactionSubmissionStatus.RevertedConfirmed
-                                : RfqmTransactionSubmissionStatus.RevertedUnconfirmed;
-                            submissionsMap[r.transactionHash].metadata = null;
-                            submissionsMap[r.transactionHash].metadata = {
-                                expectedTakerTokenFillAmount: expectedTakerTokenFillAmount.toString(),
-                                actualTakerFillAmount: '0',
-                                decodedFillLog: '{}',
-                            };
-                        }
-                        submissionsMap[r.transactionHash].blockMined = new BigNumber(r.response.blockNumber);
-                        submissionsMap[r.transactionHash].gasUsed = new BigNumber(r.response.gasUsed);
-                        submissionsMap[r.transactionHash].updatedAt = new Date();
-                    } else {
-                        submissionsMap[r.transactionHash].status = RfqmTransactionSubmissionStatus.DroppedAndReplaced;
-                        submissionsMap[r.transactionHash].blockMined = null;
-                        submissionsMap[r.transactionHash].gasUsed = null;
-                        submissionsMap[r.transactionHash].updatedAt = new Date();
-                        submissionsMap[r.transactionHash].metadata = {
-                            expectedTakerTokenFillAmount: expectedTakerTokenFillAmount.toString(),
-                            actualTakerFillAmount: '0',
-                            decodedFillLog: '{}',
-                        };
-                    }
-                }
-                await this._dbUtils.updateRfqmTransactionSubmissionsAsync(Object.values(submissionsMap));
-                if (jobStatus !== null) {
-                    await this._dbUtils.updateRfqmJobAsync(orderHash, { status: jobStatus });
-                }
-                break;
+        // No txs mined, nothing to do
+        if (minedReceipt === undefined) {
+            return {
+                isTxMined,
+                isTxConfirmed,
+                submissionsMap,
+            };
+        }
+
+        // A tx was mined
+        isTxMined = true;
+        const currentBlock = await this._blockchainUtils.getCurrentBlockAsync();
+        if (currentBlock - minedReceipt.response!.blockNumber >= BLOCK_FINALITY_THRESHOLD) {
+            isTxConfirmed = true;
+        }
+
+        // Update the SubmissionMap for this mined tx
+        if (minedReceipt.response!.status === 1) {
```

Comment:
does typescript require the declaration that response is defined here?

Reply:
Might be able to optional chain here